src/runtime/ExecutionContext.cpp

```diff
@@ -25,6 +25,11 @@
 
 namespace Escargot {
 
+FunctionObject* ExecutionContext::function()
+{
+    return m_lexicalEnvironment->record()->asDeclarativeEnvironmentRecord()->asFunctionEnvironmentRecord()->functionObject();
```

Comment:
Could `ExecutionContext::function()` be replaced by `ExecutionContext::resolveCallee()`?
If not, I think some guard codes (ASSERT) are necessary to check `isDeclarativeEnvironmentRecord()` and `isFunctionEnvironmentRecord()`

Reply:
On a second thought there are cases where this `ExecutionContext::function()` would not return the correct callee (bound functions for example) , so we must use `ExecutionContext::resolveCallee()`. Thanks for the question.